test-analysis-core/src/main/kotlin/com/saveourtool/save/test/analysis/api/TestIdGenerator.kt

```diff
@@ -42,3 +52,55 @@ fun interface TestIdGenerator {
                 DefaultTestIdGenerator()
     }
 }
+
+/**
+ * Generates a unique test id.
+ *
+ * @param organizationName the name of the organization.
+ * @param projectName the name of the project.
+ * @param testSuiteSourceName the name of the test suite source.
+ * @param testSuiteVersion the version of the test suite.
+ * @param testSuiteName the name of the test suite.
+ * @param pluginName the name of the `save-cli` plug-in ("warn", "fix", etc.).
+ * @param filePath the path to the file within the test suite.
+ * @return the generated unique test id.
+ */
+@Suppress(
+    "LongParameterList",
+    "TOO_MANY_PARAMETERS",
+)
+fun TestIdGenerator.testId(
+    organizationName: OrganizationName,
```

Comment:
why do we need all these names, if test has `id` from database?

Reply:
As discussed, extracted into a separate issue: #1726.